### DISCUSSION TITLE
Referente ao Curso Fundamentos deo Asp Net 6

### DISCUSSION THREAD

**Discussion by marcelocaselli** [balta-io/forum#385] [Tira Dúvidas] [answered]
Bom dia, lá no tópico Autenticação e Autorização, na penúltima aula Gerando Tokens com Claims, quando uso o Postman para gerar o token
![gerando tokens com claims](https://github.com/balta-io/forum/assets/111652069/e80105a4-21de-4821-a6fa-fada62fb683c)
, em vez de retornar o Token ele me retorna apenas status 200.......pelo que entendi deveria gerar o token e não é o que acontece. No código está tudo ok como conferido(o print do código está anexo).
O que pode estar havendo?
vlw

**Comment by andrebaltieri** [comment, marked as answer]
@marcelocaselli você chegou a colocar um breakpoint na linha 90 pra ver se o token está sendo gerado?

**Comment by marcelocaselli** [comment]
![Captura](https://github.com/balta-io/forum/assets/111652069/3cc617ed-5434-487b-8b80-b5497bc04db8)
nao sei se estou sabendo debugar a aplicação direito com o visual studio 2022, porém encontrei esse erro

**Reply by andrebaltieri** [reply]
@marcelocaselli este erro ocorre por que já tem alguma aplicação rodando nesta porta, você pode mudar a porta do seu App, ou reiniciar a máquina (as vezes resolve). Deve ter ficado algo no cache!

**Reply by marcelocaselli** [reply]
acabei de fazer, porém a situação persiste.
A requisição e resposta está acontecendo, só que o retorno não é o token e null(como na linha 91), e sim null e null.
Eu vi que vc fez algumas alterações no banco, será que pode ser isso?

**Reply by andrebaltieri** [reply]
@marcelocaselli como vai? Acredito que são dois problemas diferentes que estamos discutindo aqui.
A última imagem que você envio mostra um erro de execução da aplicação, isto foi resolvido com minhas dicas acima?

O segundo ponto é sobre a resposta vazia, você inspecionou a variável para ver se o Token está preenchido? Tem um print da resposta?

**Reply by marcelocaselli** [reply]
![debug3png](https://github.com/balta-io/forum/assets/111652069/f4b52f1a-3d0b-411d-8452-f1832056238b)
quando eu coloco o breakpoint na linha 90 acontece isso

**Reply by andrebaltieri** [reply]
@marcelocaselli perfeito, agora você precisa chamar o endpoint em que o Token será gerado para que a execução pare no Breakpoint.
O endereço raiz, até onde me lembro não tem resultado mesmo.

**Reply by marcelocaselli** [reply]
![postman1](https://github.com/balta-io/forum/assets/111652069/75d55daa-52e8-4a9f-8f03-f920259c1bd1)
no postman esse é o retorno

**Reply by marcelocaselli** [reply]
na aula, no data = retorna a string do token e errors = null.......no meu ta retornando os dois null

**Reply by andrebaltieri** [reply]
Bom dia, @marcelocaselli, como vai?
Durante o debug, você viu se o seu Token está preenchido e se perde quando vai para o objeto Response, ou se o Token nem está sendo preenchido?

Você precisa ir passo a passo para entender o que está ocorrendo.

**Reply by marcelocaselli** [reply]
bom dia Balta, eu fiz o debug passo a passo mas como sou leigo não consegui ter muito entendimento do debug, coloquei o breakpoint na linha 90 e ele passou, porém na linha 91 ele da erro.
Eu alterei o código da linha 91 para:     return Ok(token);
após essa alteração o token é exibido no Postman, porém no código antigo ele retorna Data = null e errors = null

**Reply by henriqueweege** [reply]
Oi @marcelocaselli, desculpa me intrometer por aqui.
Refraseando o que o professor Balta disse:  você chegou a verificar se o construtor da classe ResultViewModel está atribuindo o valor passado ao construtor para a propriedade "Data"?
Caso ainda não tenha ficado claro, a classe "ResultViewModel" deveria parecer com isso:
```
 public class ResultViewModel<T>
    {
        public ResultViewModel(T data, List<string> errors)
        {
            Data = data; //atribuição do parâmetro a propriedade
            Errors = errors;
        }

        public ResultViewModel(T data)
        {
            Data = data; //atribuição do parâmetro a propriedade
        }

        public ResultViewModel(List<string> errors)
        {
            Errors = errors;
        }

        public ResultViewModel(string error)
        {
            Errors.Add(error);
        }

        public T Data { get; private set; }
        public List<string> Errors { get; private set; } = new();
    }
```

Repare que em dois dos construtores da classe o valor recebido como parâmetro "data" é atribuído a propriedade "Data". A falta dessa atribuição pode resultar nesse comportamento de gerar o Token mas não aparecer no Postman.

Ah, sobre esse erro que ocorre na linha 91, teria como mandar um print dele?

Espero que ajude ^-^

**Reply by marcelocaselli** [reply]
cara vc tinha razão, o erro estava no primeiro construtor, Estava Data = Data.......eu troquei para Data = data e matou o problema.......cada erro simples que deixa a gente de cabelo em pé kkkk abraço vlw

**Reply by henriqueweege** [reply]
Que ótimo! fico feliz em ajudar!! 
Abração!!

**Reply by andrebaltieri** [reply]
@henriqueweege muito obrigado pela ajuda e meus parabéns pela atitude 💜

**Reply by henriqueweege** [reply]
@andrebaltieri muito obrigado! é sempre um prazer ajudar!  💜
